conn.go

```diff
@@ -140,28 +147,51 @@ func SystemBusPrivate() (*Conn, error) {
 	return Dial(defaultSystemBusAddress)
 }
 
+// SystemBusPrivateHandler returns a new private connection to the system bus, using the provided handlers.
+func SystemBusPrivateHandler(handler Handler, signalHandler SignalHandler) (*Conn, error) {
+	address := os.Getenv("DBUS_SYSTEM_BUS_ADDRESS")
+	if address != "" {
```

Comment:
(Minor tidy for consideration)
This logic could nicely be captured by a function shared by the two uses:
```golang
func getSystemBusAddress() {
    address := os.Getenv("DBUS_SYSTEM_BUS_ADDRESS")
    if address != "" {
       return address
    }
    return defaultSystemBusAddress
}
```
A similar comment applies to the session bus logic above, although I would go so far as to have it return the autolaunch address.